tests/integration/test_remove.py

```diff
@@ -0,0 +1,340 @@
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+
+import pytest
+
+from cylc.flow.commands import force_trigger_tasks, remove_tasks, run_cmd
+from cylc.flow.cycling.integer import IntegerPoint
+from cylc.flow.flow_mgr import FLOW_ALL
+from cylc.flow.scheduler import Scheduler
+from cylc.flow.task_outputs import TASK_OUTPUT_SUCCEEDED
+from cylc.flow.task_proxy import TaskProxy
+
+
+@pytest.fixture
+def example_workflow(flow):
+    return flow({
+        'scheduling': {
+            'graph': {
+                'R1': '''
+                    a1 & a2 => b
+                    a3 => b
+                ''',
+            },
+        },
+    })
+
+
+def get_data_store_flow_nums(schd: Scheduler, itask: TaskProxy):
+    _, ds_tproxy = schd.data_store_mgr.store_node_fetcher(itask.tokens)
+    if ds_tproxy:
+        return ds_tproxy.flow_nums
+
+
+async def test_basic(
+    example_workflow, scheduler, start, db_select
+):
+    """Test removing a task from all flows."""
+    schd: Scheduler = scheduler(example_workflow)
+    async with start(schd):
+        a1 = schd.pool._get_task_by_id('1/a1')
+        a2 = schd.pool._get_task_by_id('1/a2')
+        schd.pool.spawn_on_output(a1, TASK_OUTPUT_SUCCEEDED)
+        schd.pool.spawn_on_output(a2, TASK_OUTPUT_SUCCEEDED)
+        await schd.update_data_structure()
+
+        assert a1 in schd.pool.get_tasks()
+        for table in ('task_states', 'task_outputs'):
+            assert db_select(schd, True, table, 'flow_nums', name='a1') == [
+                ('[1]',),
+            ]
+        assert db_select(
+            schd, True, 'task_prerequisites', 'satisfied', prereq_name='a1'
+        ) == [
+            ('satisfied naturally',),
+        ]
+        assert get_data_store_flow_nums(schd, a1) == '[1]'
+
+        await run_cmd(remove_tasks(schd, ['1/a1'], [FLOW_ALL]))
+        await schd.update_data_structure()
+
+        assert a1 not in schd.pool.get_tasks()  # removed from pool
```

Comment:
I think the comments would be easier to read the line above the assert statements.